### ISSUE TITLE
QM frame cropped for pages with a height less than the browser's

### ISSUE THREAD

**Issue by Cipa** [ClipperCMS/ClipperCMS#303]
See the bottom of the colorbox iframe

![screen shot 2013-06-19 at 1 51 35 pm](https://f.cloud.github.com/assets/293377/676665/411ee830-d90b-11e2-9e3a-c781515ea47f.png)

**Comment by TimGS** [comment]
It seems a combination of a few things.
1) The body tag doesn't fill the viewport. No rule against that, it's not wrong, but it is relevant.
2) Colorbox appears to be using viewport size as an indicator of how big to make the 'popup' iframe.
3) qm.inc.php generates javascript to put overflow: hidden on the body tag when the popup appears.

I don't know why (3) was implemented - presumably it's for a reason. Removing this did not show any obvious problems. However I hesitate to make such a change this close to release.

Also it would appear that this issue has existed for a while without being a huge issue - presumably most sites don't have pages small enough to be a problem. I don't know which page you used for you example, I couldn't find one that was an issue on my screen - I had to delete content via Opera Dragonfly to reproduce the issue.

As such, I am wondering if we can either:
1. Ignore it till 1.3! Remove (3) above early on in 1.3 development and note any problems.
2. Set html, body { height: 100% }. At first glance this doesn't break anything on the sample site, but I've only tried on Opera. @kp52 and @microcipcip - do you think this is safe change at this point?

-- Tim.

**Comment by microcipcip** [comment]
I don't think it is a good idea to set the html and body to { height: 100% } as that could break the website design (it depend on the design). Maybe qm.inc.php is used for custom parameters?

Why don't we delete or comment line 520,521,557,558 of qm.inc.php? I don't know what's the purpose of that. Colorbox should work all right without that.

**Comment by TimGS** [comment]
re: html, body { height: 100% }. Sorry, I was unclear. This was only intended for the sample site CSS - not a general fix.

re: lines in qm.inc.php. Until someone identifies why these lines are there, we don't know what this will break! Bad idea at the point of release.

**Comment by Cipa** [comment]
Commenting the first 2 lines in the bind seem to fix the problem(qm.inc.php). I will submit a patch right away, after I test in IE

```
                                // Bindings
                                '.$jvar.'(document).bind("cbox_open", function(){
                                    //'.$jvar.'("body").css({"overflow":"hidden"});
                                    //'.$jvar.'("html").css({"overflow":"hidden"});
                                    '.$jvar.'("#qmEditor").css({"display":"none"});
                                });
```

**Comment by TimGS** [comment]
But why were these lines put there? We ought to have some idea of the functionality we are removing.

**Comment by Cipa** [comment]
Not really sure why they are there BUT a modal window should not require extra help to properly work. Maybe it was something that was needed for the old version of Colorbox. If I remember correctly I updated to the latest one earlier this year.

But it could be something else. Let me know if you find something. I will not submit the pull request right now.

**Comment by TimGS** [comment]
Maybe see what anyone else has to say. We can then decide later on.

**Comment by bossloper** [comment]
There are a few posts around about this if you google: colorbox body overflow hidden
I think it is something to do with hiding scroll bars?

**Comment by Cipa** [comment]
I googled it but they seem not to appear for us. Maybe it was for an older version. Our is from March 2013. I am 99% sure we can remove those lines. Can you do some test as well? It works on IE8, FF and Chrome, so I suspect it should work on the rest of them.

**Comment by TimGS** [comment]
@uxello - I think you might be onto something, that makes sense, but I also find @Cipa's points convincing.

**Comment by bossloper** [comment]
When I view in IE10 and Firefox with the lines commented out as suggested I get scrollbars showing for the modal site in the background:

![scrollbars](https://f.cloud.github.com/assets/1744687/681076/19e9c018-d9a7-11e2-923b-b98cb5d0bce5.jpg)

**Comment by Cipa** [comment]
@uxello - is that the latest develop? From the looks of it you might use an older version.
Can you try with a new setup. I will test in IE10

**Comment by bossloper** [comment]
yes latest develop from today (with latest clear cache on QM)... but did not overwrite default site content.

I only see the scroll bars if I comment out the two lines as suggested below, otherwise no scroll bars:
                                    //'.$jvar.'("body").css({"overflow":"hidden"});
                                    //'.$jvar.'("html").css({"overflow":"hidden"});

**Comment by Cipa** [comment]
Can you run an upgrade?

**Comment by Cipa** [comment]
No bars in IE10 for me

![screen shot 2013-06-20 at 9 16 30 am](https://f.cloud.github.com/assets/293377/681215/ae0ea160-d9ab-11e2-8a26-f7d7dbabd6ad.png)

**Comment by bossloper** [comment]
With the two lines commented out?

**Comment by Cipa** [comment]
actually no :(
I forgot to comment I get the bars in IE10 as well

**Comment by bossloper** [comment]
ok. I cannot reproduce your first post, i.e. the cropped colorbox. Which browser was that with?

**Comment by Cipa** [comment]
It happens on all browsers even IE10, Only on big screens and when the html height is smaller than the widow height. I tested it with the Contact page.

I think we can make this work by detecting the height of the browser and the height of the html. If the html height is bigger apply the overflow hidden otherwise leave as is.

PS: I was wrong about commenting those 2 lines. Actually there are 4 lines that apply the overflow: hidden and in my case the other 2 made me believe that commenting the first 2 fixed the problem.

**Comment by Cipa** [comment]
Fix here
https://github.com/ClipperCMS/ClipperCMS/pull/309

**Comment by microcipcip** [comment]
@Uxello, are you sure did you install the latest version of ClipperCMS? I can see that you have the MODX demo website installed.

@Cipa, I think you need to hide all these lines 520,521,557,558 from qm.inc.php, because the last two (557,558) are applyed to the body height when you use the button "Create Resource here", which changes the height again.

**Comment by bossloper** [comment]
@microcipcip. Absolutely latest dev, I just did not update the site content when I upgraded as I have some test pages I did not want to have deleted. I reinstalled and overwrote the demo site and now see the same issue, so it is site specific.

I tracked it down to line 418 in the demo site foundation.css.
position: relative on the body tag. If this is removed then colorbox is fine... but if there is a fix already then great.
